cli/command/service/ps.go

```diff
@@ -74,27 +74,34 @@ func runPS(dockerCli *command.DockerCli, opts psOptions) error {
 		return err
 	}
 
+loop:
 	for _, service := range opts.services {
-		serviceCount := 0
-		// Lookup by ID/Prefix
+		// Lookup by full ID match
 		for _, serviceEntry := range serviceByIDList {
-			if strings.HasPrefix(serviceEntry.ID, service) {
+			if serviceEntry.ID == service {
 				filter.Add("service", serviceEntry.ID)
-				serviceCount++
+				continue loop
 			}
 		}
 
-		// Lookup by Name/Prefix
+		// Lookup by Name
 		for _, serviceEntry := range serviceByNameList {
-			if strings.HasPrefix(serviceEntry.Spec.Annotations.Name, service) {
+			if serviceEntry.Spec.Annotations.Name == service {
 				filter.Add("service", serviceEntry.ID)
-				serviceCount++
+				continue loop
 			}
 		}
-		// If nothing has been found, return immediately.
-		if serviceCount == 0 {
-			return errors.Errorf("no such services: %s", service)
+
+		// Lookup by ID-prefix
+		for _, serviceEntry := range serviceByIDList {
+			if strings.HasPrefix(serviceEntry.ID, service) {
+				filter.Add("service", serviceEntry.ID)
+				continue loop
```

Comment:
I agree we need to have a standard solution. I was actually considering to error out if `matches > 1`; this will complicate the logic a bit, because we'll have to check number of matches per keyword, and possibly de-duplicate the keywords (e.g. `docker service ps foo foo`)